sequence_processing_pipeline/Commands.py

```diff
@@ -22,7 +22,8 @@ def split_similar_size_bins(data_location_path, max_file_list_size_in_gb,
     # is now the following:
     # add one more level to account for project_names nested under ConvertJob
     # dir.
-    fastq_paths = glob.glob(data_location_path + '*/*/*.fastq.gz')
+    # this will ignore the _I1_ reads that appear in the integrated result.
+    fastq_paths = glob.glob(data_location_path + '/*/*_R?_001.fastq.gz')
```

Comment:
I have a link to document it in another part of the source, or in qp-klp, but the short answer is part of Illumina's convention for naming output fastq files: (https://imperial-genomics-facility.github.io/igf-pipeline-help/demultiplexing.html - one of many sources). In practice the wetlab does create sample-names that include _I1_, _R2_, and others. It's very important for the glob to catch on the _right_ '_Rn_' or '_In_' value because I have seen some samples recently that had both in the same file name. The '_R?_001.fastq.gz' is actually very stable. So much so that we're renaming the output files from TellRead to match this convention so it works with downstream processes.

Reply:
Not to disagree with you but I just want to point out that this line of code is in split_similar_size_bins(), which is used exclusively by NuQCJob() which will only be used on output fastq files from bcl-convert/bcl2fastq and now TellRead. The glob is also already highly dependent on the file/directory structure <data_location_path>/<anything, presumably project-name>/something.fastq.gz. If we want to generalize this function then we need to generalize it all the way and have it accept a sorted list of fastq file paths instead. The caller will need then be responsible for sorting the list so that \_R1\_ and \_R2\_ files are adjacent, or else it will break a primary assumption on the function. This is a bug fix for a known issue.

Reply:
@antgonza I really do think that if we want to repurpose NuQCJob to accept inputs other than from ConvertJob or TellReadJob then that should be a separate issue and a separate PR, because it goes beyond the simple removal of '_001'.

Reply:
As a reminder, this morning we discussed a new functionality that will run NuQC + the rest of the steps on any raw artifact in Qiita; which doesn't enforce/follow the 001 file naming.

Reply:
Right, but I need to treat that as a separate concern. I can't delay this PR past the holidays because we want to do what I agree is only natural for the new functionality. This needs to be one PR and that needs to be another one.

Reply:
Right but this solution doesn't work for all known use cases. Note that I haven't tested this but why not do something simpler like this:

```
fastq_paths = sorted(glob.glob(data_location_path + '*/*/*.fastq.gz'))
# check if there are I# files in our folder, if we check the first that should be enough:
if '_I1_001.fastq.gz' in fastq_paths[0]:
    fastq_paths = [x for x in fastq_paths if '_I1_001.fastq.gz' not in x]
```

Reply:
It's not just I1 files. If we have a file like my_sample_name_R1.fastq.gz and my_sample_name_R2_R1.fastq.gz glob will not sort these files with their respective pair the way we assume they will.

The issue is glob itself. It's great to use it to sort R1 and R2 files so that they're adjacent in a list and you can pull them off as a pair in a loop efficiently. However if there are funky sample-names sort is not going to sort them in the correct order.

Reply:
I don't know any scenario where we will have:
```
my_sample_name_R1.fastq.gz and my_sample_name_R2_R1.fastq.gz 
```
we might have:
```
my_sample_name_R1.fastq.gz and my_sample_name_R1_R1.fastq.gz 
my_sample_name_R2.fastq.gz and my_sample_name_R2_R1.fastq.gz 
```
thus, the propose solution will still work, no?

Reply:
I'm just going to change it so it will be easier to use with the new feature you mentioned this morning. Kudos :)